pkg/nettest/nettest.go

```diff
@@ -0,0 +1,30 @@
+// Package nettest provides functionality to run tests that access the public network.
+package nettest
+
+import (
+	"os"
+	"sync"
+	"testing"
+)
+
+var m sync.Mutex
```

Comment:
If the issue is that tests **from different packages** run concurrently, this won't fix it - each package to test gets compiled separately, so each will get their own mutex.

Reply:
Huh, even if you run them as `go test [...] ./cmd/... ./ee/... ./orbit/... ./pkg/... ./server/... ./tools/...`?

Reply:
(you can see it in action if you want by running `$ watch pgrep -l .test` in a terminal window and starting e.g. `go test ./server/...` in a separate one)

Reply:
Oh true! TIL.

Assuming we don't want to over-engineer this, how about using Redis (that we already start to run the tests) to implement the locking... I would also only enable this "behavior" via an env variable to run on CI only.

Reply:
:+1: for CI only I think that approach makes sense. You should still use a TTL on the key even though it shouldn't matter on CI, just in case we enable that env var locally from time to time (to ensure that the lock is removed - whatever happens - after a while).

Reply:
A single file lock might do it? (Just pushed). And it's something to use on CI only, so I added a `TEST_LOCK_FILE_PATH` var to enable it.